app/javascript/packages/memorable-date/index.ts

```diff
@@ -0,0 +1,386 @@
+/**
+ * Keys to lookup error messages for different states
+ * of the MemorableDate element
+ */
+export enum MemorableDateErrorMessage {
+  missing_month_day_year = 'missing_month_day_year',
```

Comment:
That being said, the more I look through how we're using this, I don't think this could be a type instead of an enum, since we're not referencing these values directly, and an enum would generate runtime code that would increase the size of the bundles unnecessarily.

I would expect something like this:

```ts
type MemorableDateErrorMessage =
  'missingMonthDayYear' |
  'missingMonthDay'
  // ...
```

Reply:
I could take a look at converting this into a [`const enum`](https://www.typescriptlang.org/docs/handbook/enums.html#const-enums).

Reply:
I refactored usages to permit converting this to a `const enum`.